lib/ocpp/common/websocket/websocket_tls_tpm.cpp

```diff
@@ -398,24 +349,65 @@ void WebsocketTlsTPM::client_loop() {
 
     info.fd_limit_per_thread = 1 + 1 + 1;
 
-    bool use_tpm = connection_options.use_tpm_tls;
+    // Setup context - need to know the key type first
+
+    std::string path_key;
+    std::string path_chain;
+    std::optional<std::string> password;
 
-    // Setup context
-    OSSL_LIB_CTX* lib_ctx;
-    SSL_CTX* ssl_ctx;
+    if (this->connection_options.security_profile == 3) {
 
-    create_sec_context(use_tpm, lib_ctx, ssl_ctx);
+        const auto certificate_key_pair =
+            this->evse_security->get_key_pair(CertificateSigningUseEnum::ChargingStationCertificate);
 
-    // Connection acquire the contexts
-    conn_data->sec_lib_context = std::unique_ptr<OSSL_LIB_CTX>(lib_ctx);
-    conn_data->sec_context = std::unique_ptr<SSL_CTX>(ssl_ctx);
+        if (!certificate_key_pair.has_value()) {
+            EVLOG_AND_THROW(std::runtime_error(
+                "Connecting with security profile 3 but no client side certificate is present or valid"));
+        }
+
+        path_chain = std::string(certificate_key_pair.value().certificate_path.c_str());
+        path_key = std::string(certificate_key_pair.value().key_path.c_str());
```

Comment:
Couldn't you just use
```suggestion
        path_chain = certificate_key_pair.value().certificate_path;
        path_key = certificate_key_pair.value().key_path;
```
in this case?

Reply:
updated